Assets/Scripts/Enemies/BossBehaviorController.cs

```diff
@@ -4,108 +4,156 @@
 using UnityEngine;
 using System.Linq;
 
-[RequireComponent (typeof(EnemyHealth))]
-public class BossBehaviorController : MonoBehaviour {
+public class BossAction : MonoBehaviour
```

Comment:
its an interface not a struct

Reply:
well its a class... not an interface (idk why i said structure instead of class because struct is an actual thing so saying (data) structure is confusing even if technically correct), interfaces cant describe what data is held in the classes which implement them, only give a guarantee of what methods exist on them etc